rust/src/lib/ifaces/ovs.rs

```diff
@@ -810,7 +810,7 @@ impl MergedInterfaces {
         for merged_iface in self.iter_mut().filter(|i| {
             if let Some(Interface::OvsBridge(o)) = i.desired.as_ref() {
                 o.base.state == InterfaceState::Up
-                    && o.bridge.as_ref().map(|c| c.allow_extra_patch_ports)
+                    && o.bridge.as_ref().and_then(|c| c.allow_extra_patch_ports)
```

Comment:
I would assume using `and_then` to avoid nested `Some()`.

Reply:
The `map` need the function inside to return T, while `and_then` need the function inside to return `Option<T>`. If that is your question.